connector/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SparkConnectStreamHandler.scala

```diff
@@ -49,6 +50,16 @@ class SparkConnectStreamHandler(responseObserver: StreamObserver[ExecutePlanResp
         .getOrCreateIsolatedSession(v.getUserContext.getUserId, v.getSessionId)
         .session
     session.withActive {
+
+      // Add debug information to the query execution so that the jobs are traceable.
+      val debugString = v.toString
```

Comment:
`v.toString` is problematic in some cases, see https://github.com/apache/spark/pull/40607/files#r1159434752

I think we have to manually implement it.

Reply:
> My guess is that you're submitting a very large Python UDF that requires lots of heap space in the string

[this one](https://github.com/apache/spark/blob/e81c2c895a66f00888198f615a1f11b410bb7475/python/pyspark/ml/torch/tests/test_distributor.py#L57-L122) , it will be converted to a Pandas UDF, should be very large

I think it is also related to java version, since it throw OOM in Java8 but works well in Java11.